docs/proposed/psa-driver-interface.md

```diff
@@ -560,6 +562,29 @@ The core will not update the persistent state in storage while an entry point is
 
 In a multithreaded environment, the driver may only call these two functions from the thread that is executing the entry point.
 
+#### Built-in keys
+
+Opaque drivers may declare built-in keys. Built-in keys can be accessed, but not created, through the PSA Cryptography API.
+
+A built-in key is identified by its location and its **slot number**. Drivers that support built-in keys must provide a `"get_builtin_key"` entry point to retrieve the key data and metadata. This entry point has the following prototype:
+
+```
+psa_status_t acme_get_builtin_key(psa_drv_slot_number_t slot_number,
+                                  psa_key_attributes_t *attributes,
+                                  uint8_t *key_buffer,
+                                  size_t key_buffer_size);
+```
+
+If this function returns `PSA_SUCCESS`, it must fill `attributes` with the attributes of the key (except for the key identifier) and must fill `key_buffer` with the key data.
+
+On entry, `psa_get_key_lifetime(attributes)` is the location at which the driver was declared and the persistence level `#PSA_KEY_LIFETIME_PERSISTENT`. The driver entry point may change the lifetime to one with the same location but a different persistence level.
+
+The output parameter `key_buffer` points to a writable buffer of `key_buffer_size` bytes. If the driver has a [`"builtin_key_size"` property](#key-format-for-opaque-drivers) property, `key_buffer_size` has this value, otherwise `key_buffer_size` has the value determined from the key type and size.
```

Comment:
This is a good question. Either we need an entry point for that, or we need to bound the size of built-in keys.

Reply:
Maybe get_built_in_key_attributes(slot_number, &attributes) and get_built_in_key_context(slot_number, key_buffer, key_buffer_size) ?

Reply:
Or maybe mandate that `get_builtin_key(slot_number, &attributes, NULL, 0)` for a valid slot number return `PSA_ERROR_BUFFER_TOO_SMALL` with correct attributes, which would allow the core to calculate the buffer size?

Reply:
In some sense, this is quite identical I'd say. In one case we have two entry points, in the other one only one entry point but a return in error for a valid and expected call. Not sure about what we gain in trying to fit the two needed queries into a single entry point.

Reply:
I went with `get_builtin_key(slot_number, &attributes, NULL, 0)`. I think having a single entry point, and making it bail out midway if the output buffer is too small, is simpler both for the core implementer and for the driver implementer.

Reply:
This makes the job, fine by me.